lib/ansible/modules/web_infrastructure/wildfly.py

```diff
@@ -0,0 +1,356 @@
+    elif result.get('outcome') == 'success':
+        if not module.check_mode:
+            undeploy_result, info = undeploy(module)
+            if info['status'] == 200:
+                return True
+            else:
+                module.fail_json(msg='Undeploy failed.', details=info['msg'])
+        else:
+            return True
+
+
+def deploy_enable(result, module):
+    if result.get('outcome') == 'failed':
+        if not module.check_mode:
+            deployment_result, info = deploy(module)
+            if info['status'] == 200:
+                return True
+            else:
+                module.fail_json(msg='Deployment failed.', details=info['msg'])
+        else:
+            return True
+    elif result.get('outcome') == 'success':
+        if not module.check_mode:
+            undeploy_result, info = undeploy(module)
+            if info['status'] == 200:
+                deployment_result, info = deploy(module)
+                if info['status'] == 200:
+                    return True
+                else:
+                    module.fail_json(
+                        msg='Deployment failed.', details=info['msg'])
+        else:
+            return True
+    else:
+        module.fail_json(msg='Deployment failed.', details=info['msg'])
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            src=dict(type='str', required=False),
+            deployment=dict(type='str', required=True),
+            url_username=dict(required=True, aliases=['username']),
+            url_password=dict(
+                required=True, aliases=['password'], no_log=True),
+            state=dict(choices=['absent', 'present'], default='present'),
+            mode=dict(choices=['domain', 'standalone'], default='domain'),
+            controller_address=dict(default='localhost:9990'),
+            server_groups=dict(type='str', default='main-server-group'),
+        ),
+        supports_check_mode=True
+    )
+
+    changed = False
+
+    state = module.params['state']
+    deployment = module.params['deployment']
+
+    if not HAS_JSON:
+        module.fail_json(msg='Cannot import JSON package.')
+
+    deployed, info = is_deployed(module)
+
+    if info['status'] == 200:
+        result = json.loads(deployed.read())
+    elif info['status'] == 500:
+        result = json.loads(info['body'])
+    else:
+        module.fail_json(msg=info['status'], details=info['msg'])
+
+    if 'outcome' in result:
+        if state == 'absent':
+            changed = deploy_disable(result, module)
+            if changed:
+                module.exit_json(
+                    changed=changed, status='undeployed', name=deployment)
+            else:
+                module.exit_json(
+                    changed=changed, status='None', name=deployment)
+        else:
+            changed = deploy_enable(result, module)
+            module.exit_json(
+                changed=changed, status='deployed', name=deployment)
```

Comment:
You are not returning `state` in any of the `exit_json`. You should drop the `status` as that can be determined from the `state` and the `changed` values.